### PR TITLE
8280076: Unify IGV and IR printing

### PR DESCRIPTION
**Pull request by neliasso**
This patch adds support for controlling printing of the textual ideal representation (PrintIdeal) in the same way as the printing of IGV graphs currently works. The new flag PrintIdealLevel uses the same phases and levels as the PrintIdealGraphLevel flag. This new flag also gets added to CompileCommand and CompilerDirectives. 

The motivation for this change is to add support for testing during different phases with the TestIR framework.

The old flag (PrintIdeal) is kept intact for now.

This patch is based upon https://github.com/openjdk/jdk/pull/7088

Please review,
Nils

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8280076](https://bugs.openjdk.java.net/browse/JDK-8280076): Unify IGV and IR printing

### Reviewers
 * [Christian Hagedorn](https://openjdk.java.net/census#chagedorn) (@chhagedorn - **Reviewer**)
 * [Tobias Hartmann](https://openjdk.java.net/census#thartmann) (@TobiHartmann - **Reviewer**)
 * [Vladimir Ivanov](https://openjdk.java.net/census#vlivanov) (@iwanowww - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/7108/head:pull/7108` \
`$ git checkout pull/7108`

Update a local copy of the PR: \
`$ git checkout pull/7108` \
`$ git pull https://git.openjdk.java.net/jdk pull/7108/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 7108`

View PR using the GUI difftool: \
`$ git pr show -t 7108`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/7108.diff">https://git.openjdk.java.net/jdk/pull/7108.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/opto/c2_globals.hpp [hunk 1]
```diff
@@ -112,6 +112,11 @@
   notproduct(bool, PrintIdeal, false,                                       \
           "Print ideal graph before code generation")                       \
                                                                             \
+  notproduct(uintx, PrintIdealLevel, 0,                                     \
+          "Print ideal IR on stdout."                                       \
```
Comment (chhagedorn) [inline src/hotspot/share/opto/c2_globals.hpp:116]: Missing space after `stdout.`.

**Review by chhagedorn — APPROVED** [review]
Nice! That's a good basis for adding IR matching on different phases to the IR framework.

File: src/hotspot/share/opto/compile.cpp [hunk 2]
```diff
@@ -535,6 +535,26 @@ void Compile::print_compile_messages() {
 #endif
 }
 
+#ifndef PRODUCT
+void Compile::print_ideal_ir(const char *name) {
```
[the remark this answers is not in the document]
Comment (neliasso) [reply src/hotspot/share/opto/compile.cpp:539]: I also removed default value in hpp declaration.

**Review by chhagedorn — APPROVED** [review]

**Review by TobiHartmann — APPROVED** [review]
Looks good to me.

**Review by iwanowww — APPROVED** [review]
Looks good.

**Pull request closed without merging**